ls1intum/Artemis
# `Programming exercises`: Export files embedded with HTML syntax

### Checklist
#### General


#### Server

### Motivation and Context


The export of a programming exercise doesn't contain files embedded using HTML syntax because the regex doesn't match them.

### Description

Added a second regex to also match files embedded with HTML syntax refactored some code to reuse logic for both regex expressions.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming exercise

1. Embedded some files in the problem statement using both markdown and html syntax
2. Export (Download) the exercise
3. Verify that the export contains all embedded files.


### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage





#### Server

| Class/File | Line Coverage | Confirmation (assert/expect) |Comment|
|------------|--------------:|---------------------:|--------------------:|
| [ProgrammingExerciseExportService.java](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5418/latest/artifact/shared/Coverage-Report-Server-Tests/de.tum.in.www1.artemis.service.programming/ProgrammingExerciseExportService.html) | 86% | ✅ | my changes are covered

### Screenshots
